### ISSUE TITLE
Empty set in scaffold split

### ISSUE THREAD

**Issue by my-alaska** [scikit-fingerprints/scikit-fingerprints#227]
It seems that there is a bug in scaffold split implementation.

For `CHEMBL1000710` data file from [FS-Mol](https://github.com/microsoft/FS-Mol) dataset, the `scaffold_train_valid_test_split() ` function returns empty training set. 
The file contains 81 molecules. 
When splitting in 70-15-15 manner, the function returns indices of size 0, 63 and 18 for train, validation, and test set respectively. 

I checked some other files to see if it's the only case, and for the same configuration, some splits are still incorrectly balanced. 
For `CHEMBL1000621` the 40 molecules get split into sets of size 20, 14 and 6 instead of 28, 6 and 6.

**Comment by StaniszewskiA** [comment]
Hi, I've added a new empty subset check in this PR **https://github.com/scikit-fingerprints/scikit-fingerprints/pull/231**. I hope it will solve the issue.

**Comment by StaniszewskiA** [comment]
Hi, I believe the issue lies in **desired_test_size** and **desired_valid_size** from **scaffold_split**. I forgot that casting a float to an integer results in taking the floor of it, so in some cases, we end up with 0 instead of the proper value. The changes in PR https://github.com/scikit-fingerprints/scikit-fingerprints/pull/234 should be enough to fix this bug
